code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	effect_desc = "creates a rune that constantly wet itself with water as long as the rune is up"
+
+/obj/effect/warped_rune/cyanspace
+	icon_state = "slipperyspace"
+	desc = "Water drowns, water feeds, water slips, is there a thing water cannot do?"
+
+/obj/effect/warped_rune/cyanspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/cyanspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slippery_rune(T)
+
+
+///just wet the turf under the rune every now and then, there's actually a small moment where the tile isn't wet
+/obj/effect/warped_rune/cyanspace/proc/slippery_rune(turf/open/T)
+	T.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+
+
+
+
+/*      Metal rune : makes an invisible wall. actually I lied, the rune is the wall.*/
+
+
+/obj/item/slimecross/warping/metal
+	colour = "metal"
+	runepath = /obj/effect/warped_rune/metalspace
+	effect_desc = "Creates a rune that prevents passage above it, takes longer to store and draw than other runes."
+	storing_time = 200  //storing time is longer for this particular rune
+	drawing_time = 100  //it's also longer to draw it
+
+//It's a wall what do you want from me
+/obj/effect/warped_rune/metalspace
+	desc = "Words are powerful things, they can stop someone dead in their tracks if used in the right way"
+	icon_state = "metal_space"
+	density = TRUE
+
+
+
+
+
+
+
+
+/*  Yellow rune space acts as an infinite generator, works without power and anywhere, recharges the APC of the room it's in and any batteries put on it
+
+*/
+
+
+/obj/item/slimecross/warping/yellow
+	colour = "yellow"
+	runepath = /obj/effect/warped_rune/yellowspace
+	effect_desc = "Creates a rune that Infinitely recharge batteries that are put on it, it will also passively recharge the APC of the room"
+
+
+/obj/effect/warped_rune/yellowspace
+	desc = "Where does all this energy come from? Who knows,the process does not matter, only the result."
+	icon_state = "elec_rune"
+
+
+/obj/effect/warped_rune/yellowspace/Initialize()
+	.=..()
+	START_PROCESSING(SSmachines, src)
+
+
+/obj/effect/warped_rune/yellowspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		apc_charge()
+		electrivore()
+
+/obj/effect/warped_rune/yellowspace/proc/apc_charge()
+
+	var/area/A = get_area(T)
+	for(var/obj/machinery/power/apc/P in A)
+		if(!P.cell)
+			return
+		if(P.cell.charge <! (P.cell.maxcharge - 1))
+			return
+		P.cell.charge += P.cell.maxcharge/100 //will basically recharge 1% of the APC cell every 10 seconds,we don't want to run engineers out of buisness
+		
+
+///charge whatever battery is put on the rune
+/obj/effect/warped_rune/yellowspace/proc/electrivore()
+	for(var/obj/item/stock_parts/cell/C in T)
+		if(C.charge < C.maxcharge) //check if the battery is not full
+			C.charge += C.maxcharge/10
+
+
+
+
+
```

Comment:
and that concludes Part II of my review

Part III is coming soon

Reply:
Thanks! I'll try to fix most of the issues you've talked about. I know it's probably a lot of work reviewing all the crossbreed so I appreciate it.

Reply:
Nah, I haven't done nearly as much work as you have on this PR.

Despite my many comments, I really do like what you've done here in this PR, and I think that the runes presented in this PR can perhaps make up for the massive loss of fun in the xenobio department that came with the removal of change carps (from xenobio).